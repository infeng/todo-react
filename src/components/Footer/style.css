.filter {
  padding: 5px;
  border: none;
  background-color: transparent;
}
.filter.selected {
  border: 1px solid rgba(175, 47, 47, 0.2);
  border-radius: 3px;
}