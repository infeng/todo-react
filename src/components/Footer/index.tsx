import * as React from 'react';
import * as filters from '../../constants/filters';

import './style.css';

const FILTER_TITLE = {
  [filters.SHOW_ALL]: 'All',
  [filters.SHOW_COMPLETED]: 'Completed',
  [filters.SHOW_ACTIVE]: 'Active',
};

interface Props {
  onFilterChange: (filter: string, name?: string) => any;
  filter: string;
}

export default class Footer extends React.Component<Props, any> {
  static propType = {
    filter: React.PropTypes.oneOf([
      filters.SHOW_ALL,
      filters.SHOW_COMPLETED,
      filters.SHOW_ACTIVE,
    ]).isRequired,
  };
  constructor() {
    super();
    this.renderFilter = this.renderFilter.bind(this);
  }

  renderFilter(filter: string, name: string) {
    if (filter === this.props.filter) {
      return (<button className="filter selected">{name}</button>);
    }
    return (
      <button className="filter" onClick={e => {
        e.preventDefault();
        this.props.onFilterChange(filter);
      }}>
        {name}
      </button>
    );
  }

  render() {
    return (
      <div>
        {this.renderFilter(filters.SHOW_ALL, FILTER_TITLE[filters.SHOW_ALL])}
        {' '}
        {this.renderFilter(filters.SHOW_COMPLETED, FILTER_TITLE[filters.SHOW_COMPLETED])}
        {' '}
        {this.renderFilter(filters.SHOW_ACTIVE, FILTER_TITLE[filters.SHOW_ACTIVE])}
      </div>
    );
  }
}
