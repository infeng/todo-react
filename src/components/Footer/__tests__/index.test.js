import React from 'react';
import { render, mount } from 'enzyme';
import Footer from '..';

function createFooter(props) {
  return (
    <Footer {...props}/>
  );
}

describe('Footer', () => {
  it('render Footer correctly', () => {
    let footer = createFooter();

    let wrapper = render(footer);
    expect(wrapper).toMatchSnapshot();
  });

  it('select filter', () => {
    let onFilterChange = jest.fn();

    let wrapper = mount(createFooter({onFilterChange}));
    
    wrapper.find('button').at(1).simulate('click');
    expect(onFilterChange).toBeCalledWith('SHOW_COMPLETED');

    wrapper.setProps({filter: 'SHOW_ACTIVE'});
    expect(wrapper.find('button').at(2).hasClass('selected')).toBe(true);
  });
});