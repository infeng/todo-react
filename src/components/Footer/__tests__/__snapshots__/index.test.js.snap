// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer render Footer correctly 1`] = `
<div>
  <button
    class="filter"
  >
    All
  </button>
   
  <button
    class="filter"
  >
    Completed
  </button>
   
  <button
    class="filter"
  >
    Active
  </button>
</div>
`;
